[TASK-2_1]: Homework Java Week 2

tayareut opened the pull request:
- Add answers for questions and test Chapter 1.
- Add answers for questions and test Chapter 2.
- Make tasks from Chapter 1, except B9.

tayareut committed:
[TASK-2_1]: Homework Java Week 2
- Add answers for questions and test Chapter 1.
- Add answers for questions and test Chapter 2.
- Make tasks from Chapter 1, except B9.

asyafedorova commented on TASK-2.1/src/chapter_1_task/a2/PrintArgumentBack.java:
```diff
@@ -0,0 +1,9 @@
+package chapter_1_task.a2;
+public class PrintArgumentBack {
+    public static void main(String[] args) {
+        for (int i = args.length - 1; i >= 0; --i) {
```
А здесь по какой-то конкретной причине --i, а не i--?

tayareut replied:
нет)

asyafedorova commented on TASK-2.1/src/chapter_1_task/a3/LineBreak.java:
```diff
@@ -0,0 +1,13 @@
+package chapter_1_task.a3;
+public class LineBreak {
+    public static void main(String[] args) {
+        int[] a = {1, 2, 3, 4, 5};
```
Не совсем правильно понято задание. На вход должно прийти количество случайных чисел, а тебе уже нужно их сгенерировать и образить в нужном виде)

tayareut replied:
package chapter_1_task.a3;
public class LineBreak {
    public static void main(String[] args) {
        for (String arg : args) {
            System.out.println(arg);
        }
        for (String arg : args) {
            System.out.print(args[0] + ", " + arg);
        }
    }
}

asyafedorova commented on TASK-2.1/src/chapter_1_task/a4/PasswordCheck.java:
```diff
@@ -0,0 +1,16 @@
+        Scanner scan = new Scanner(System.in);
+        String password1 = scan.next();
+        String password2 = "sample";
+        if (password1.equals(password2)){
```
В if-else потерялись пробелы

tayareut replied:
done

asyafedorova commented on TASK-2.1/src/chapter_1_task/a5/NumbersActions.java:
```diff
@@ -0,0 +1,14 @@
+    public static void main (String [] args){
+        System.out.println("Ввести 2 целых числа: ");
+        Scanner scan = new Scanner(System.in);
+        int x = scan.nextInt();
```
Это не аргументы командной строки)

tayareut replied:
package chapter_1_task.a5;
public class NumbersActions {
    public static void main (String [] args){
        int sum = 0;
        int multi = 1;
        for (String arg : args) {
            int stringToInt = Integer.valueOf(arg);
            sum += stringToInt;
            multi *= stringToInt;
        }
        System.out.println("Сумма аргументов: " + sum);
        System.out.println("Произведение аргументов: " + multi);
    }
}

asyafedorova commented on TASK-2.1/src/chapter_1_task/b10/FindPalindromicNumbers.java:
```diff
@@ -0,0 +1,32 @@
+        }
+    }
+
+    public static boolean isPalindromic(int nums) {
```
Огонь, что сделала отдельный метод!

asyafedorova commented on TASK-2.1/src/chapter_1_task/b10/FindPalindromicNumbers.java:
```diff
@@ -0,0 +1,32 @@
+        String word = String.valueOf(nums);
+        int length = word.length();
+        for (int i = 0; i < (length / 2); i++) {
+            if (word.charAt(i) != word.charAt(length - i - 1)) {
```
Можно сделать лучше: перевести строку в массив чаров. Тогда не придётся каждый раз дёргать charAt

asyafedorova commented on TASK-2.1/src/chapter_1_task/b2/MaxMin.java:
```diff
@@ -0,0 +1,22 @@
+        for (int i = 0; i < size; i++) {
+            nums[i] = scanner.nextInt();
+        }
+        OptionalInt max = Arrays.stream(nums).max();
```
Хитро) А если ручками проверять без встроенных функций?)

asyafedorova replied:
Для алгоритмов "Грокаем алгоритма"

asyafedorova commented on TASK-2.1/src/chapter_1_task/b5/ThreeDigitNumbers.java:
```diff
@@ -0,0 +1,26 @@
+            nums[i] = scanner.nextInt();
+        }
+        System.out.println("Все  трехзначные  числа,  в  десятичной  записи  которых  нет  одинаковых цифр:");
+        for (int num : nums) {
```
Надо подумать над рациональностью решений

asyafedorova commented on TASK-2.1/src/chapter_1_task/b6/PrimeNumbers.java:
```diff
@@ -0,0 +1,30 @@
+                System.out.println(nums[i]);
+        }
+    }
+        public static boolean isPrime(int size){
```
Ненужный отступ

asyafedorova commented on TASK-2.1/src/chapter_1_task/b7/SortedNumbers.java:
```diff
@@ -0,0 +1,34 @@
+        nums[i] = scanner.nextInt();
+    }
+    System.out.println("Отсортированные числа в порядке возрастания:");
+        Arrays.sort(nums);
```
Хитро) Хочу ручками сортировку

asyafedorova reviewed (commented):
Added comments

asyafedorova reviewed (commented):
Added comments

tayareut committed:
[TASK-2_1]: Update Homework Java Week 2
- Implement all comments, except b5

asyafedorova commented on TASK-2.1/src/chapter_1_task/b2/MaxMin.java:
```diff
@@ -14,9 +12,22 @@ public static void main(String[] args) {
-        System.out.println("Максимальное число:" + max.getAsInt());
-        OptionalInt min = Arrays.stream(nums).min();
-        System.out.println("Минимально число:" + min.getAsInt());
+        int MaxValue = 0;
```
Нейминг переменной неправильный

tayareut committed:
[TASK-2_1]: Homework Java Week 2 Update 2
- Implement all comments to a4, b2, b5.
- b9 Lucky numbers - it's still not clear.
- Add b1, b2 from Chapter 2, tried to make separate classes.

tayareut committed:
[TASK-2_1]: Homework Java Week 2 Update 3
- Remove static methods.
- Make Main method in one file.
- Add task b3

tayareut committed:
[TASK-2_1]: Homework Java Week 2 Update 4
- Take all methods from Main into a separate class WorkingClass.java.
- Edit HomeWork_2.java
- Add task b4

tayareut committed:
[TASK-2_1]: Homework Java Week 2 Update 4
- Implement comments to:
CountUniqueNumbers.java
IndexOfMinElementInArray.java
WorkingClass.java

- Add tasks a5, a6, a7, a8
- Add HomeWork_2.java with tasks b1, b2, b3
